sdk/plugin_test.go

```diff
@@ -844,3 +845,22 @@ func TestListenSettings_Validate_Error(t *testing.T) {
 		assert.Equal(t, testCase.errCount, len(merr.Errors), merr.Error())
 	}
 }
+
+// TestRedactedJsonConfigOutput tests that we redact passwords when we dump
+// them to the logs. map[string]interface{} values will be replaced with
+// REDACTED if the lowercase map key containes "pass".
+func TestRedactedJsonConfigOutput(t *testing.T) {
+	// Test string from a log before we redacted passwords.
+	s := "{\"Version\":\"1\",\"Debug\":true,\"Settings\":{\"Mode\":\"serial\",\"Listen\":{\"Enabled\":true,\"Buffer\":100},\"Read\":{\"Enabled\":true,\"Interval\":\"3s\",\"Buffer\":1024},\"Write\":{\"Enabled\":true,\"Interval\":\"1s\",\"Buffer\":1024,\"Max\":100},\"Transaction\":{\"TTL\":\"5m\"},\"Cache\":{\"Enabled\":false,\"TTL\":180000000000}},\"Network\":{\"Type\":\"tcp\",\"Address\":\"0.0.0.0:5003\",\"TLS\":null},\"DynamicRegistration\":{\"Config\":[{\"authenticationPassphrase\":\"Vap0rIsDriven!\",\"authenticationProtocol\":\"SHA\",\"endpoint\":\"10.193.3.201\",\"model\":\"PXGMS UPS + EATON 93PM\",\"port\":161,\"privacyPassphrase\":\"Vap0rIsDriven!\",\"privacyProtocol\":\"AES\",\"userName\":\"user\",\"version\":\"v3\"}]},\"Limiter\":null,\"Health\":{\"UseDefaults\":true},\"Context\":{}}"
```

Comment:
Can we change the values in here with test data? I don't want to be leaking any IPs or passwords in this public repo.

Reply:
This is the production reproduction case and is a dead password.

Reply:
All the same, the password may be used elsewhere and I don't think we need that exact password, dead or not, in here to test the functionality.